### DISCUSSION TITLE
Problem with normalization in MarkerFeatures

### DISCUSSION THREAD

**Discussion by anastasiya-pendragon** [GreenleafLab/ArchR#1362] [Questions / Documentation]
Hello there! First of all, thank you so much for this amazing package! It makes analysis way easier, especially for beginners like me.

I've noticed something in the MarkerFeatures function, that seems wrong to me. It concerns normalization. 
https://github.com/GreenleafLab/ArchR/blob/506161d36d1457d4b3aba20918c0e4f622ae937e/R/MarkerFeatures.R#L206-L226
Specifically lines 223 and 224. As it is right now, it seems that if normBy is set to any character value and is not "none", the same normalization would be done, no matter what was specified as the desired normalization column. I tried to run it with different normBy parameters (without changing anything else) and the results are always the same. I would think that maybe these lines simply need to be replaced by 
```
normFactors <- getCellColData(ArchRProj, normBy, drop=FALSE)            
normFactors[,1] <- median(normFactors[,1]) / normFactors[,1]
```
Or maybe there has to be an additional check on what matrix and/or test is used. 

I checked both issues and discussions and the same code snippet was talked about here: https://github.com/GreenleafLab/ArchR/discussions/1168#discussioncomment-1662307, but in a different context, so I thought this might deserve a new discussion. 

If I'm misinterpreting or missing something in the code, please let me know. 
The code snippet is from the 1.0.2 branch but it is the same in previous versions. 

For now, my workaround has been to set ReadsInTSS to nFrags and normBy to null, and after the analysis undo the changes. 

Thank you for your time! 

Looking forward to hearing from you, 
Anastasiya

**Comment by rcorces** [comment]
moving this to Issues
